ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs

```diff
@@ -35,66 +35,92 @@ private void OpenFilexml()
-                            .SelectMany(d => d.Elements())
-                            .Select(e => e.Name.ToString())
-                            .Distinct();
-
-                var headers =
-                    String.Join(",",
-                        getFields(xml)
-                            .Select(f => csvFormat(f)));
-
-                var query =
-                    from dealer in xml.Descendants("Content").Descendants("attendees")
-                    select string.Join(",",
-                        getFields(xml)
-                            .Select(f => dealer.Elements(f).Attributes("Text").Any()
-                                ? dealer.Element(f).Value
-                                : "")
-                            .Select(x => csvFormat(x)));
-
-                var csv =
-                    String.Join(Environment.NewLine,
-                        new[] { headers }.Concat(query));
-
-                SaveFileDialog saveFileDialog1 = new SaveFileDialog();
-                saveFileDialog1.Filter = "CSV files (*.csv)|*.csv";
-                DialogResult dr = saveFileDialog1.ShowDialog();
-                if (dr == DialogResult.OK)
+                try
                 {
-                    string filename = saveFileDialog1.FileName;
-                    File.WriteAllText(filename, csv);
-                    MessageBox.Show("Save File Success");
-                    wr.Close();
+                    StreamReader wr = new StreamReader(textBox1.Text, true);
+                    string fs = wr.ReadToEnd();
 
-                }
+                    Func<string, string> csvFormat =
+          t => String.Format("\"{0}\"", t.Replace("\"", "\"\""));
 
-            }
+                    var xml = XDocument.Parse(fs);
+
+                    Func<XDocument, IEnumerable<string>> getFields =
+                        xd =>
+                            xd
+                                .Descendants("Content")
+                                 .Elements()
+                                .SelectMany(d => d.Elements())
+                                .Select(e => e.Name.ToString())
+                                .Distinct();
+
+
+                    var headers = String.Join(",", getFields(xml).Select(f => csvFormat(f)));
+
+                    var header = "";
+                    List<string> name = new List<string>();
+                    foreach (var head in headers.Split(','))
+                    {
+                        if (ContainsText(head))
+                        {
+                            name.Add(head);
+
+                        }
+                    }
+
+                    header = String.Join(",", name);
 
-        }
 
-        //public static List<Dictionary<string, string>>WritingInCSV(string absolutePath)
-        //{
+                    var query =
+                        from dealer in xml.Descendants("Content").Elements()
+                        select string.Join(",",
```

Comment:
Hmm, here you are creating CSV lines, like `"value1","value2","value3"`?
This is what `CsvHelper` should do for you. 
You should be passing an enumerable of objects to `CsvHelper`, like `csv.WriteRecords( records );`, or if you aren't using a class, using `csv.WriteField(x)`. 

It looks like you want to do the writing twice, in [l106](https://github.com/salween/ToolsConvert/pull/4/files#diff-30f1d823aa8372641e7a88bdee2e9ca0R106) and again in [XmltoCsv l22](https://github.com/salween/ToolsConvert/pull/4/files#diff-c880c42188cd6e578dacf416c056180dR22), but I'm not sure I understand what is happening there. It's preparing a writer but not using it.